### PR TITLE
add unit test for pkg/resourceinterpreter/default/native/dependencies.go

### PR DESCRIPTION
**Pull request by yike21**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
The remaining task of #3339
add unit test for `pkg/resourceinterpreter/default/native/dependencies.go`

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

```release-note

```

### CODE REVIEW

**Comment by yike21** [comment]
Please find the test report below:
```shell
Running tool: /usr/local/bin/go test -timeout 30s -run ^(Test_getDeploymentDependencies|Test_getJobDependencies|Test_getCronJobDependencies|Test_getPodDependencies|Test_getDaemonSetDependencies|Test_getStatefulSetDependencies)$ github.com/karmada-io/karmada/pkg/resourceinterpreter/default/native

=== RUN   Test_getDeploymentDependencies
=== RUN   Test_getDeploymentDependencies/deployment_without_dependencies
=== PAUSE Test_getDeploymentDependencies/deployment_without_dependencies
=== RUN   Test_getDeploymentDependencies/deployment_with_dependencies_1
=== PAUSE Test_getDeploymentDependencies/deployment_with_dependencies_1
=== RUN   Test_getDeploymentDependencies/deployment_with_dependencies_2
=== PAUSE Test_getDeploymentDependencies/deployment_with_dependencies_2
=== RUN   Test_getDeploymentDependencies/deployment_with_dependencies_3
=== PAUSE Test_getDeploymentDependencies/deployment_with_dependencies_3
=== CONT  Test_getDeploymentDependencies/deployment_without_dependencies
=== CONT  Test_getDeploymentDependencies/deployment_with_dependencies_2
=== CONT  Test_getDeploymentDependencies/deployment_with_dependencies_3
=== CONT  Test_getDeploymentDependencies/deployment_with_dependencies_1
--- PASS: Test_getDeploymentDependencies (0.00s)
    --- PASS: Test_getDeploymentDependencies/deployment_without_dependencies (0.00s)
    --- PASS: Test_getDeploymentDependencies/deployment_with_dependencies_3 (0.00s)
    --- PASS: Test_getDeploymentDependencies/deployment_with_dependencies_2 (0.00s)
    --- PASS: Test_getDeploymentDependencies/deployment_with_dependencies_1 (0.00s)
=== RUN   Test_getJobDependencies
=== RUN   Test_getJobDependencies/job_without_dependencies
=== PAUSE Test_getJobDependencies/job_without_dependencies
=== RUN   Test_getJobDependencies/job_with_dependencies_1
=== PAUSE Test_getJobDependencies/job_with_dependencies_1
=== RUN   Test_getJobDependencies/job_with_dependencies_2
=== PAUSE Test_getJobDependencies/job_with_dependencies_2
=== RUN   Test_getJobDependencies/job_with_dependencies_3
=== PAUSE Test_getJobDependencies/job_with_dependencies_3
=== CONT  Test_getJobDependencies/job_without_dependencies
=== CONT  Test_getJobDependencies/job_with_dependencies_2
=== CONT  Test_getJobDependencies/job_with_dependencies_1
=== CONT  Test_getJobDependencies/job_with_dependencies_3
--- PASS: Test_getJobDependencies (0.00s)
    --- PASS: Test_getJobDependencies/job_with_dependencies_2 (0.00s)
    --- PASS: Test_getJobDependencies/job_with_dependencies_3 (0.00s)
    --- PASS: Test_getJobDependencies/job_with_dependencies_1 (0.00s)
    --- PASS: Test_getJobDependencies/job_without_dependencies (0.00s)
=== RUN   Test_getCronJobDependencies
=== RUN   Test_getCronJobDependencies/cronjob_without_dependencies
=== PAUSE Test_getCronJobDependencies/cronjob_without_dependencies
=== RUN   Test_getCronJobDependencies/cronjob_with_dependencies_1
=== PAUSE Test_getCronJobDependencies/cronjob_with_dependencies_1
=== RUN   Test_getCronJobDependencies/cronjob_with_dependencies_2
=== PAUSE Test_getCronJobDependencies/cronjob_with_dependencies_2
=== RUN   Test_getCronJobDependencies/cronjob_with_dependencies_3
=== PAUSE Test_getCronJobDependencies/cronjob_with_dependencies_3
=== CONT  Test_getCronJobDependencies/cronjob_without_dependencies
=== CONT  Test_getCronJobDependencies/cronjob_with_dependencies_2
=== CONT  Test_getCronJobDependencies/cronjob_with_dependencies_1
=== CONT  Test_getCronJobDependencies/cronjob_with_dependencies_3
--- PASS: Test_getCronJobDependencies (0.00s)
    --- PASS: Test_getCronJobDependencies/cronjob_without_dependencies (0.00s)
    --- PASS: Test_getCronJobDependencies/cronjob_with_dependencies_2 (0.00s)
    --- PASS: Test_getCronJobDependencies/cronjob_with_dependencies_1 (0.00s)
    --- PASS: Test_getCronJobDependencies/cronjob_with_dependencies_3 (0.00s)
=== RUN   Test_getPodDependencies
=== RUN   Test_getPodDependencies/pod_without_dependencies
=== PAUSE Test_getPodDependencies/pod_without_dependencies
=== RUN   Test_getPodDependencies/pod_with_dependencies_1
=== PAUSE Test_getPodDependencies/pod_with_dependencies_1
=== RUN   Test_getPodDependencies/pod_with_dependencies_2
=== PAUSE Test_getPodDependencies/pod_with_dependencies_2
=== RUN   Test_getPodDependencies/pod_with_dependencies_3
=== PAUSE Test_getPodDependencies/pod_with_dependencies_3
=== CONT  Test_getPodDependencies/pod_without_dependencies
=== CONT  Test_getPodDependencies/pod_with_dependencies_2
=== CONT  Test_getPodDependencies/pod_with_dependencies_1
=== CONT  Test_getPodDependencies/pod_with_dependencies_3
--- PASS: Test_getPodDependencies (0.00s)
    --- PASS: Test_getPodDependencies/pod_with_dependencies_1 (0.00s)
    --- PASS: Test_getPodDependencies/pod_with_dependencies_3 (0.00s)
    --- PASS: Test_getPodDependencies/pod_with_dependencies_2 (0.00s)
    --- PASS: Test_getPodDependencies/pod_without_dependencies (0.00s)
=== RUN   Test_getDaemonSetDependencies
=== RUN   Test_getDaemonSetDependencies/daemonset_without_dependencies
=== PAUSE Test_getDaemonSetDependencies/daemonset_without_dependencies
=== RUN   Test_getDaemonSetDependencies/daemonset_with_dependencies_1
=== PAUSE Test_getDaemonSetDependencies/daemonset_with_dependencies_1
=== RUN   Test_getDaemonSetDependencies/daemonset_with_dependencies_2
=== PAUSE Test_getDaemonSetDependencies/daemonset_with_dependencies_2
=== RUN   Test_getDaemonSetDependencies/daemonset_with_dependencies_3
=== PAUSE Test_getDaemonSetDependencies/daemonset_with_dependencies_3
=== CONT  Test_getDaemonSetDependencies/daemonset_without_dependencies
=== CONT  Test_getDaemonSetDependencies/daemonset_with_dependencies_2
=== CONT  Test_getDaemonSetDependencies/daemonset_with_dependencies_3
=== CONT  Test_getDaemonSetDependencies/daemonset_with_dependencies_1
--- PASS: Test_getDaemonSetDependencies (0.00s)
    --- PASS: Test_getDaemonSetDependencies/daemonset_without_dependencies (0.00s)
    --- PASS: Test_getDaemonSetDependencies/daemonset_with_dependencies_1 (0.00s)
    --- PASS: Test_getDaemonSetDependencies/daemonset_with_dependencies_2 (0.00s)
    --- PASS: Test_getDaemonSetDependencies/daemonset_with_dependencies_3 (0.00s)
=== RUN   Test_getStatefulSetDependencies
=== RUN   Test_getStatefulSetDependencies/statefulset_without_dependencies
=== PAUSE Test_getStatefulSetDependencies/statefulset_without_dependencies
=== RUN   Test_getStatefulSetDependencies/statefulset_with_dependencies_1
=== PAUSE Test_getStatefulSetDependencies/statefulset_with_dependencies_1
=== RUN   Test_getStatefulSetDependencies/statefulset_with_dependencies_2
=== PAUSE Test_getStatefulSetDependencies/statefulset_with_dependencies_2
=== RUN   Test_getStatefulSetDependencies/statefulset_with_dependencies_3
=== PAUSE Test_getStatefulSetDependencies/statefulset_with_dependencies_3
=== CONT  Test_getStatefulSetDependencies/statefulset_without_dependencies
=== CONT  Test_getStatefulSetDependencies/statefulset_with_dependencies_2
=== CONT  Test_getStatefulSetDependencies/statefulset_with_dependencies_1
=== CONT  Test_getStatefulSetDependencies/statefulset_with_dependencies_3
--- PASS: Test_getStatefulSetDependencies (0.00s)
    --- PASS: Test_getStatefulSetDependencies/statefulset_with_dependencies_1 (0.00s)
    --- PASS: Test_getStatefulSetDependencies/statefulset_without_dependencies (0.00s)
    --- PASS: Test_getStatefulSetDependencies/statefulset_with_dependencies_3 (0.00s)
    --- PASS: Test_getStatefulSetDependencies/statefulset_with_dependencies_2 (0.00s)
PASS
ok      github.com/karmada-io/karmada/pkg/resourceinterpreter/default/native    0.023s


> 测试运行完成时间: 2023/4/11 18:30:41 <


```

**Comment by XiShanYongYe-Chang** [comment]
Thanks a lot~

> Please find the test report below:

I'm sorry for misleading you with my previous statement, you don't need to state the results of the ut test in pr, we can get it in ci.

**Comment by yike21** [comment]
> Thanks a lot~
> 
> > Please find the test report below:
> 
> I'm sorry for misleading you with my previous statement, you don't need to state the results of the ut test in pr, we can get it in ci.

I understand now, thanks!

File: pkg/resourceinterpreter/default/native/dependencies_test.go [hunk 1]
```diff
@@ -1 +1,847 @@
...
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	configv1alpha1 "github.com/karmada-io/karmada/pkg/apis/config/v1alpha1"
+)
+
+var (
+	namespace = "karmada-test"
+	podSpec1  = &unstructured.Unstructured{
```
Comment (XiShanYongYe-Chang) [inline pkg/resourceinterpreter/default/native/dependencies_test.go:14]: Can you help change `podSpec1` to a more readable name? Same as other parts.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~
/lgtm
/approve

**Review by RainbowMango — APPROVED** [review]
/approve
Thanks.

**Pull request merged**